src/features/capabilities/capability.ts

```diff
@@ -0,0 +1,23 @@
+import { Client } from "../../client";
+import { apiLocation } from "../../apiLocation";
+import { OctopusError } from "../../octopusError";
+
+interface CapabilitiesResponse {
+    Capabilities: string[];
+}
+
+export async function checkForCapability(client: Client, capabilityName: string): Promise<string | null> {
+    try {
+        const response = await client.get<CapabilitiesResponse>(`${apiLocation}/capabilities`);
+        return response.Capabilities.filter((c) => c === capabilityName).length === 1
+            ? null
+            : `The Octopus instance does not support ${capabilityName}, you may need to upgrade it to get access to the feature you are trying to use.`;

```

Comment:
extend this message to include a link/callout to support, and the same for the messages below